lib/server.ts

```diff
@@ -220,18 +220,23 @@ export function createApp(devJsonPath: string, opts: DevOptions): Express.Applic
    */
   app.all('/api/proxy-rpc', proxyMiddleware(RPC_URL[opts.network]));
 
-  if (opts.gateway) {
+  if (gateway.enabled) {
     log.debug("Setting up gateway...");
     if (opts.index) {
 
+			// use near-bos-webcomponent locallyl
+			// use as gateway the dist bundle in index.html;
+			// change something like bacjkgrind color;
+			// check if the example here reflect the changes;
+			// this is a test to understand if -g/bos.config.json gateway optino works;
+
+
       log.debug("Index provided. Using new gateway setup.");
       // use new path
-      let gatewayUrl = typeof opts.gateway === 'string' ? opts.gateway : DEFAULT_REMOTE_GATEWAY_URL;
-      const isLocalPath = !gatewayUrl.startsWith('http');
-      gatewayUrl = gatewayUrl.replace(/\/$/, ''); // remove trailing slash
-      opts.gateway = gatewayUrl; // standardize to url string
 
-      initializeGateway(gatewayUrl, isLocalPath, opts, devJsonPath);
+      const isLocalPath = !gateway.bundleUrl.startsWith('http');
+
+      const gatewayInitPromise = initializeGateway(gateway, isLocalPath, opts, devJsonPath);
```

Comment:
I think this `const gatewayInitPromise =` would introduce strange behavior? Does the web server stutter at all when starting?

I thought gatewayInitPromise is a constant already defined in this file and initializeGateway function was setting it

In other words, I don't think you should have this const gatewayInitPromise here